### ISSUE TITLE
[Improvement][API] Deduplicate logs in controllers

### ISSUE THREAD

**Issue by haydenzhourepo** [apache/dolphinscheduler#5098]
**Describe the question**
there are so many logs in controllers, those logs just want log who request which method with what params and headers, 

![image](https://user-images.githubusercontent.com/21101605/111597240-f1a26600-8808-11eb-9388-6e343116c4a6.png)

**What are the current deficiencies and the benefits of improvement**
so many repetitive visit logs! we can handle those logs by aop， so do not have to write them in each controller。
and don't worry about forgetting.

**Which version of DolphinScheduler:**
 -[1.3.6]

**Describe alternatives you've considered**
use aop handle controller layer logs unified

**Comment by chengshiwen** [comment]
@haydenzhourepo Would you like to submit the pr to improve this issue? Welcome to participate in the DolphinScheduler opensource community.

**Comment by ruanwenjun** [comment]
I think we can add a general audit function in the aspect. It is better to record the request params and response body, it can help us to troubleshoot.

**Comment by haydenzhourepo** [comment]
@chengshiwen I will submit a pr for this. there are three solutions:
1. use a filter to log the request and response
2. use interceptor to log the request and response
3. use aspect to log the request and response

filter is based on servlet container, and others are based on reflection.
so I think filter is the best choice for our case, What do you think of this.

**Comment by kalencaya** [comment]
a little warn about filter.
I had developed same request log functions within dubbo and springmvc, which dubbo for aspect and springmvc for filter.
Filter worked well for springmvc after I solved response stream can't be readed many times by ContentCachingRequestWrapper and ContentCachingResponseWrapper.
But when I tried to add async http api, new question encountered. springmvc framework supports Callable、WebAsyncTask、DeferredResult async methods all failed, which can't be responsed until timeout. Only servlet3  AsyncContext succeed just I operated response stream directly. 
when I replaced filter with AsyncHandlerInterceptor provided by springmvc framework,  above two questions disappeared.

hope my experience can help you.
